### ISSUE TITLE
Shellcheck fails to parse when variables expand to operators

### ISSUE THREAD

**Issue by dcsobral** [koalaman/shellcheck#1645]
#### For bugs
- SC1072
- 0.6.0
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
A=1
B=-lt
C=2
[ "$A" $B "$C" ] && echo yes || echo no
```

#### Here's what shellcheck currently says:
```
In demo.sh line 5:
[ "$A" $B "$C" ] && echo yes || echo no
^-- SC1073: Couldn't parse this test expression. Fix to allow more checks.
       ^-- SC1072: Expected test to end here (don't wrap commands in []/[[]]). Fix any mentioned problems and try again.

For more information:
  https://www.shellcheck.net/wiki/SC1072 -- Expected test to end here (don't ...
  https://www.shellcheck.net/wiki/SC1073 -- Couldn't parse this test expressi...
```

#### Here's what I wanted or expected to see:
Maybe warnings related to unquoted variables, but no errors.

**Comment by matthewpersico** [comment]
Yes. I have tried this too. I don't remember what the disposition of my issue entry was. I will have to search it.

**Comment by matthewpersico** [comment]
Cannot seem to find any issue I authored on the subject of dynamic operators.

**Comment by Nightfirecat** [comment]
Maybe related to #1382?

**Comment by dcsobral** [comment]
Looks to be the same, really. It's annoying, because that particular test
is used on OSHT, a unit testing library.

-- 
Daniel C. Sobral

**Comment by graywolf-at-work** [comment]
The wiki page https://www.shellcheck.net/wiki/SC1072 is not much useful. Maybe the workaround of using test could be documented there?
